Content.Server/_DV/Surgery/SurgeryCleanSystem.cs

```diff
@@ -0,0 +1,53 @@
+using Content.Shared._DV.Surgery;
+using Content.Shared.Forensics;
+using Content.Shared.FixedPoint;
+using System.Linq;
+
+namespace Content.Server._DV.Surgery;
+
+/// <summary>Responsible for handling the visual appearance of and sanitzation of items that can get dirty from surgery</summary>
+public sealed class SurgeryCleanSystem : SharedSurgeryCleanSystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SurgeryDirtinessComponent, SurgeryCleanedEvent>(OnCleanDirt);
+        SubscribeLocalEvent<SurgeryCrossContaminationComponent, SurgeryCleanedEvent>(OnCleanDNA);
+    }
+
+    public override bool RequiresCleaning(EntityUid target)
+    {
+        var isDirty = (TryComp<SurgeryDirtinessComponent>(target, out var dirtiness) && dirtiness.Dirtiness > 0);
+        var isContaminated = (TryComp<SurgeryCrossContaminationComponent>(target, out var contamination) && contamination.DNAs.Count > 0);
+
+        return isDirty || isContaminated;
+    }
+
+    private void OnCleanDirt(Entity<SurgeryDirtinessComponent> ent, ref SurgeryCleanedEvent args)
+    {
+        ent.Comp.Dirtiness = FixedPoint2.Max(FixedPoint2.Zero, ent.Comp.Dirtiness - args.DirtAmount);
+        Dirty(ent);
+    }
+
+    private void OnCleanDNA(Entity<SurgeryCrossContaminationComponent> ent, ref SurgeryCleanedEvent args)
+    {
+        for (int i = 0; i < args.DnaAmount; i++)
+        {
+            if (ent.Comp.DNAs.Count <= 0)
+                break;
+
+            ent.Comp.DNAs.Remove(ent.Comp.DNAs.ElementAt(0));
+        }
+
```

Comment:
use RemoveWhere and increment a counter instead of this, removing if the counter is below the event dna count